test/client_ui_test.py

```diff
@@ -84,6 +85,111 @@ def test_draw():
     assert f.last == line.ljust(79) + "\n"
 
 
+def test_json_format():
+    fake_time = FakeTime()
+    f = FakeFile()
+
+    # Size is unknown at this point.
+    pb = client.ProgressBar(
+        phase="setting up", output=f, format="json", now=fake_time)
+    line_data = {
+        "transferred": 0,
+        "size": None,
```

Comment:
This is likey to cause issues for users, when they will be surprised to get null value. Not getting any value may be less surprising, and the caller never have to deal with invalid values.